### ISSUE TITLE
Add support for No-Op Value

### ISSUE THREAD

**Issue by tnovotny** [WhiZTiM/UbjsonCpp#6]
If I read the source code correctly then the [No-Op](http://ubjson.org/type-reference/value-types/#noop) value is not considered. This is especially important for the streams.

**Comment by tnovotny** [comment]
just saw the
`enum class Marker : byte   { ...   No_Op   = 'N',`
so I think this can just be closed

**Comment by WhiZTiM** [comment]
You are right, while we do have the `Marker::No_Op` marker, I didn't consider it in the `StreamReader`. `StreamWriter` not having it can be an acceptable temporary limitation. However, it is definitely a bug that `StreamReader` doesn't have it. (What if we are reading UBJSON binary stream written by another client?).

Thanks! Added to my TODO list this weekend.
